src/packages/builder/src/codegen/index.ts

```diff
@@ -1,114 +1,113 @@
+import { Source } from '@graphql-tools/utils';
 
 type CodegenOptions = {
 	typesOutputPath?: string | string[];
 };
 
+type CodegenConfig = { [key: string]: any };
+
 const content = `/* eslint-disable */
 /* 
 * This file is auto-generated by Graphweaver. 
 * Please do not edit it directly.
 */`;
 
-export const codeGenerator = async (schema: string, options?: CodegenOptions) => {
-	try {
-		const files = await executeCodegen({
-			cwd: process.cwd(),
-			pluginLoader: async (plugin: string) => import(plugin),
-			schema,
-			ignoreNoDocuments: true,
-			documents: ['./src/**/!(*.generated).{ts,tsx}'],
-			generates: {
-				'src/types.generated.ts': {
-					config: {
-						skipDocumentsValidation: {
-							skipDuplicateValidation: true, // A flag to disable the validation for duplicate query and mutation names we don't need this as we are using near-operation-file
-						},
-					},
-					plugins: [
-						{
-							add: {
-								content,
-							},
-						},
-						'typescript',
-					],
-				},
-				'src/': {
-					preset: nearOperationFilePreset,
-					presetConfig: {
-						extension: '.generated.ts',
-						baseTypesPath: 'types.generated.ts',
-					},
-					plugins: [
-						{
-							add: {
-								content,
-							},
-						},
-						'typescript-operations',
-						'typescript-react-apollo',
-					],
-				},
-			},
-		});
+const generateCombinedTypeFiles = async (
+	config: CodegenConfig,
+	schema: DocumentNode,
+	documents: Source[]
+) => {
+	const typesFileContent = await codegen({
+		schema,
+		documents,
+		filename: 'src/types.generated.ts',
+		config,
+		plugins: [{ typescriptPlugin: {} }],
+		pluginMap: {
+			typescriptPlugin: typescript,
+		},
+	});
+	fs.writeFileSync('src/types.generated.ts', content.concat('\n', typesFileContent));
+	fs.writeFileSync('src/frontend/types.ts', content.concat('\n', typesFileContent));
+};
 
-		const typesOutputPaths = formatListOfTypeOutputPaths(options?.typesOutputPath);
+const generateIndividualTypeFiles = async (
+	config: CodegenConfig,
+	schema: DocumentNode,
+	documents: Source[]
+) => {
+	const argsList = await nearOperationFilePreset.buildGeneratesSection({
+		presetConfig: {
+			extension: '.generated.ts',
+			baseTypesPath: 'types.generated.ts',
+		},
+		plugins: [{ typescriptOperationsPlugin: {} }, { typescriptReactApolloPlugin: {} }],
+		schema,
+		documents,
+		config,
+		pluginMap: {
+			typescriptOperationsPlugin: typescriptOperations,
+			typescriptReactApolloPlugin: typescriptReactApollo,
+		},
+		baseOutputDir: '.',
+	});
 
-		// Write the files to disk
-		const writeOperations = files.flatMap((file) => [
-			fs.promises.writeFile(path.join(file.filename), file.content, 'utf8'),
-			// We save the types to two locations src and .graphweaver / outdir
-			...(file.filename === 'src/types.generated.ts'
-				? typesOutputPaths.map((outputPath: string) =>
-						fs.promises.writeFile(outputPath, file.content, 'utf8')
-				  )
-				: []),
-		]);
+	for (const args of argsList) {
```

Comment:
Do we need this loop twice?

Looks like we are looping here and writing the file and then looping again on line 71?

Reply:
Correct - deleted